### ISSUE TITLE
Highlighting confused by single quotes in non-expr variable assignments

### ISSUE THREAD

**Issue by kylebrandt** [tbutterwith/bosun-vscode#5]
Many of alerts have a `$notes` variables that then is referenced in a template. When there is a single quote in these the parser gets confused:

![selection_111](https://user-images.githubusercontent.com/1692624/51751977-38aacb80-2084-11e9-8bcc-1bfb84ada36f.jpg)

Text interpolation ... I just **don't** know ;-)

If it helps, variable assignments in the expression language areas *can not* span multiple lines, even with something back tick quotes. So once you reach the end of the line in an expression variable assignment you can abandon whatever (in something, e.g. quotes) state you in.

**Comment by tbutterwith** [comment]
Thanks for highlighting @kylebrandt

**Comment by kylebrandt** [comment]
@tbutterwith Thanks! Updated my version and tested on our prod config, looks resolved now. FYI I started to play with some gists https://gist.github.com/kylebrandt/126557dffc9b5fc72d15bd2620034138 .

Since I believe these are both textmate under the hood, if I ever get time to redo and finish my https://github.com/bosun-monitor/bosun/pull/2044 branch I should be able to convert by changing some property names and joining the array.
